Mathlib/MeasureTheory/Decomposition/Exhaustion.lean

```diff
@@ -0,0 +1,335 @@
+    exact measure_spanningSets_lt_top _ _
+  · simp only [Nat.pairEquiv_symm_apply, f, e]
+    rw [← Set.union_iUnion]
+    suffices ⋃ n, (spanningSets (μ.restrict (μ.sigmaFiniteSetGE ν (Nat.unpair n).1)) n.unpair.2
+        ∩ μ.sigmaFiniteSetGE ν n.unpair.1) = μ.sigmaFiniteSetWRT' ν by
+      rw [this, Set.compl_union_self]
+    calc ⋃ n, (spanningSets (μ.restrict (μ.sigmaFiniteSetGE ν (Nat.unpair n).1)) n.unpair.2
+        ∩ μ.sigmaFiniteSetGE ν n.unpair.1)
+      = ⋃ n, ⋃ m, (spanningSets (μ.restrict (μ.sigmaFiniteSetGE ν n)) m
+            ∩ μ.sigmaFiniteSetGE ν n) :=
+          Set.iUnion_unpair (fun n m ↦ spanningSets (μ.restrict (μ.sigmaFiniteSetGE ν n)) m
+            ∩ μ.sigmaFiniteSetGE ν n)
+    _ = ⋃ n, μ.sigmaFiniteSetGE ν n := by
+        refine Set.iUnion_congr (fun n ↦ ?_)
+        rw [← Set.iUnion_inter, iUnion_spanningSets, Set.univ_inter]
+    _ = μ.sigmaFiniteSetWRT' ν := rfl
+
+/-- `μ.sigmaFiniteSetWRT' ν` has maximal `ν`-measure among all measurable sets `s` with sigma-finite
+`μ.restrict s`. -/
+lemma measure_sigmaFiniteSetWRT' (μ ν : Measure α) [IsFiniteMeasure ν] :
+    ν (μ.sigmaFiniteSetWRT' ν)
+      = ⨆ (s) (_ : MeasurableSet s) (_ : SigmaFinite (μ.restrict s)), ν s := by
+  apply le_antisymm
+  · refine (le_iSup (f := fun _ ↦ _)
+      (inferInstance : SigmaFinite (μ.restrict (μ.sigmaFiniteSetWRT' ν)))).trans ?_
+    exact le_iSup₂ (f := fun s _ ↦ ⨆ (_ : SigmaFinite (μ.restrict s)), ν s) (μ.sigmaFiniteSetWRT' ν)
+      (measurableSet_sigmaFiniteSetWRT' μ ν)
+  · exact le_of_tendsto' (tendsto_measure_sigmaFiniteSetGE μ ν)
+      (fun _ ↦ measure_mono (Set.subset_iUnion _ _))
+
+/-- For all measurable sets `s` in `(μ.sigmaFiniteSetWRT ν)ᶜ`, if `ν s ≠ 0` then `μ s = ∞`. -/
+lemma measure_eq_top_of_subset_compl_sigmaFiniteSetWRT'
+    (hs : MeasurableSet s) (hs_subset : s ⊆ (μ.sigmaFiniteSetWRT' ν)ᶜ) (hνs : ν s ≠ 0) :
+    μ s = ∞ := by
+  suffices ¬ SigmaFinite (μ.restrict s) by
+    by_contra h
+    have h_lt_top : Fact (μ s < ∞) := ⟨Ne.lt_top h⟩
+    exact this inferInstance
+  intro hsσ
+  have h_lt : ν (μ.sigmaFiniteSetWRT' ν) < ν (μ.sigmaFiniteSetWRT' ν ∪ s) := by
+    rw [measure_union _ hs]
+    · exact ENNReal.lt_add_right (measure_ne_top _ _) hνs
+    · exact disjoint_compl_right.mono_right hs_subset
+  have h_le : ν (μ.sigmaFiniteSetWRT' ν ∪ s) ≤ ν (μ.sigmaFiniteSetWRT' ν) := by
+    conv_rhs => rw [measure_sigmaFiniteSetWRT']
+    refine (le_iSup
+      (f := fun (_ : SigmaFinite (μ.restrict (μ.sigmaFiniteSetWRT' ν ∪ s))) ↦ _) ?_).trans ?_
+    · infer_instance
+    · exact le_iSup₂ (f := fun s _ ↦ ⨆ (_ : SigmaFinite (μ.restrict _)), ν s)
+        (μ.sigmaFiniteSetWRT' ν ∪ s) ((measurableSet_sigmaFiniteSetWRT' μ ν).union hs)
+  exact h_lt.not_le h_le
+
+end IsFiniteMeasure
+
+section SFinite
+
+lemma measure_eq_top_of_subset_compl_sigmaFiniteSetWRT [SFinite ν]
+    (hs : MeasurableSet s) (hs_subset : s ⊆ (μ.sigmaFiniteSetWRT ν)ᶜ) (hνs : ν s ≠ 0) :
```

Comment:
If my comment above works fine, then you should be able to remove `hs` here, and then the lemma ` measure_eq_top_of_absolutelyContinuous_of_subset_compl_sigmaFiniteSetWRT` becomes redundant, right?